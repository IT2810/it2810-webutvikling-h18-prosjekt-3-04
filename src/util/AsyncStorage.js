import { AsyncStorage } from "react-native";

//Retrieve all tasks
export async function RetrieveTodos() {
    try {
        return await AsyncStorage.getItem('todos', null);
    } catch (error) {
        alert(error);
    }
}

//Add all tasks
export async function StoreTodos(todos) {
    try {
        await AsyncStorage.setItem('todos', JSON.stringify(todos));
    } catch (error) {
        alert(error);
    }
}

//Add one task
export async function AddTodo(todo) {
    try {
        const todoString = await RetrieveTodos();
        let todos;

        if (todoString === null) { todos = JSON.parse('[]') }
        else { todos = JSON.parse(todoString); }

        todos.unshift(todo);
        await AsyncStorage.setItem('todos', JSON.stringify(todos));
    } catch (error) {
        alert(error);
    }
}

//Remove all tasks
export async function Clear() {
    try {
        await AsyncStorage.clear();
    } catch (error) {
        alert(error);
    }
}