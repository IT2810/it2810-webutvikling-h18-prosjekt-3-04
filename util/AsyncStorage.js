import { AsyncStorage } from "react-native";

export async function AddInitialTodos() {
    try {
        let initialList = [
                {
                    data: 'This is a test',
                    deadline: '19.10.2018',
                    type: 'text'
                },
                {
                    data: 'This is also a test',
                    deadline: 'No deadline',
                    type: 'text'
                },
        ];

        await AsyncStorage.setItem('todos', JSON.stringify(initialList));
        console.log('Added initial todos');
    } catch (error) {
        alert(error);
    }
}

export async function RetrieveTodos() {
    try {
        return await AsyncStorage.getItem('todos', null);
    } catch (error) {
        alert(error);
    }
}

export async function AddTodo(todo) {
    try {
        const todoString = await RetrieveTodos();
        let todos;

        if (todoString === null) { todos = JSON.parse('[]') }
        else { todos = JSON.parse(todoString); }

        todos.push(todo);
        await AsyncStorage.setItem('todos', JSON.stringify(todos));
    } catch (error) {
        alert(error);
    }
}

export async function RemoveTodo(index) {
    try {
        const todoString = await RetrieveTodos();
        let todos = JSON.parse(todoString);
        todos.splice(index, 1);
        //console.log(todos);
        await AsyncStorage.setItem('todos', JSON.stringify(todos));
    } catch (error) {
        alert(error);
    }
}

export async function Clear() {
    try {
        await AsyncStorage.clear();
    } catch (error) {
        alert(error);
    }
}