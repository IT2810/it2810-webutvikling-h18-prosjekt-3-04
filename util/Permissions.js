import { Permissions } from 'expo';

let permissionMap = {
    'cameraRoll' : Permissions.CAMERA_ROLL,
    'camera' : Permissions.CAMERA,
    'location' : Permissions.LOCATION,
    'notifications' : Permissions.NOTIFICATIONS,
    'audioRecording' : Permissions.AUDIO_RECORDING,
    'reminders' : Permissions.REMINDERS,
    'brightness' : Permissions.SYSTEM_BRIGHTNESS,
};

/**
 * @return {boolean}
 */

export async function RequestPermission(permission) {
    const response = await Permissions.askAsync(permissionMap[permission]);
    //console.log(response);
    if (response['status'] !== 'granted') {
        alert('Permission ' + permission + ' not granted. Permissions can be changed in settings');
        return false;
    }
    return true;
}

/**
 * @return {boolean}
 */

export async function CheckPermission(permission) {
    const { status } = await Permissions.getAsync(permissionMap[permission]);
    return status === 'granted';
}
