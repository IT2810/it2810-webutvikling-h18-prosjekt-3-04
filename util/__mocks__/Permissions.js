const mockedModule = jest.mock('RequestPermission');

//mockedModule.RequestPermission = jest.fn();

module.exports = mockedModule;