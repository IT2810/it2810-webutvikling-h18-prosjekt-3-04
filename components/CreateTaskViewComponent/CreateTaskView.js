import React from 'react';
import { TouchableOpacity, Text , View } from 'react-native';
import styles from './styles/styles'
import DateTimePicker from 'react-native-modal-datetime-picker';
import { AddTodo } from "../../util/AsyncStorage";
import ChooseTextComponent from '../ChooseTextComponent/ChooseTextComponent'
import ChooseImageComponent from '../ChooseImageComponent/ChooseImageComponent'
import UploadImageModalComponent from '../UploadImageModalComponent/UploadImageModalComponent'
import { RequestPermission } from "../../util/Permissions";
import { ImageManipulator, ImagePicker } from "expo";


export default class CreateTaskView extends React.Component {

    constructor(props) {
        super(props);
        this._pickImage = this._pickImage.bind(this);
        this._pickCameraImage = this._pickCameraImage.bind(this);
        this._toggleModal = this._toggleModal.bind(this);
        this._postTask = this._postTask.bind(this);
        this._updateText = this._updateText.bind(this);
        this._toggleDatePicker = this._toggleDatePicker.bind(this);
        this._handleDatePicked = this._handleDatePicked.bind(this);
        this.state = {
            isDatePickerVisible: false,
            isModalVisible: false,
            currentTask: {
                type: 'text',
                data: null,
                deadline: 'Whenever you want',
                checked: false,
            }
        };
    }

    static navigationOptions = {
        title: 'Create New Task',
        headerStyle: {
            backgroundColor: '#fff'
        },
        headerTintColor: '#ff0042',
        headerTitleStyle: {
            color: '#ff0042',
            fontSize: 16,
            fontWeight: "bold",
        }
    };

    componentDidMount() {
        let tempTask = this.state.currentTask;
        tempTask.type = this.props.navigation.getParam('type', 'text');
        this.setState({ currentTask: tempTask });
    }

    _postTask = async() => {
        if (this.state.currentTask.data === null || this.state.currentTask.data === '') {
            alert('Cannot add empty task');
            return;
        }

        await AddTodo(this.state.currentTask);
        this.props.navigation.navigate('Home');
    };

    _toggleDatePicker = () => { this.setState(prevState => ({ isDatePickerVisible: !prevState.isDatePickerVisible }))};

    _toggleModal = () => { this.setState(prevState => ({ isModalVisible: !prevState.isModalVisible }))};

    _handleDatePicked = (date) => {
        let deadlineString = date.toString().substring(0,16) + 'at ' + date.toString().substring(16,21);
        console.log('Deadline: ' + deadlineString);

        let tempTask = this.state.currentTask;
        tempTask.deadline = deadlineString;
        this.setState({ currentTask: tempTask });
        this._toggleDatePicker();
    };

    _updateText(text) {
        let tempTask = this.state.currentTask;
        tempTask.data = text;
        this.setState({ currentTask: tempTask });
    }

    _pickImage = async() => {
        const status = await RequestPermission('cameraRoll');

        if (!status) {
            return;
        }

        const result = await ImagePicker.launchImageLibraryAsync({
            allowsEditing: true,
            base64: false,
        });

        this._updateImage(result);
    };

    _pickCameraImage = async() => {
        const statusCamera = await RequestPermission('camera');
        const statusLibrary = await RequestPermission('cameraRoll');

        if (!statusCamera || !statusLibrary) {
            return;
        }

        const result = await ImagePicker.launchCameraAsync({
            allowsEditing: true,
            base64: false,
        });

        this._updateImage(result);
    };

    _updateImage = async(result) => {
        if (result["cancelled"]) {
            return;
        }

        const uri = result.uri;
        const actions = [{ resize: { width:300 } }];
        const saveOptions = {
            compress: 0.5,
            format: 'jpeg',
            base64: true,
        };

        const newImage = await ImageManipulator.manipulate(uri, actions, saveOptions);
        let tempTask = this.state.currentTask;
        tempTask.data = newImage.base64;
        this._toggleModal();
        this.setState({ currentTask: tempTask });
    };

    renderTypeSpecificComponents() {
        if (this.state.currentTask.type === 'image') {
            return (
                <React.Fragment>
                    <ChooseImageComponent currentTask={this.state.currentTask} toggleModal={this._toggleModal}/>
                    <UploadImageModalComponent
                        isModalVisible={this.state.isModalVisible}
                        pickImage={this._pickImage}
                        pickCameraImage={this._pickCameraImage}
                        toggleModal={this._toggleModal}/>
                </React.Fragment>
            )
        } else {
            return <ChooseTextComponent updateText={this._updateText}/>;
        }
    }

    render() {

        return (
            <View style={styles.container}>
                {this.renderTypeSpecificComponents()}
                <TouchableOpacity style={styles.deadlineBtn} onPress={this._toggleDatePicker}>
                    <Text style={styles.buttonText}>{this.state.currentTask.deadline === 'Whenever you want' ?
                        'Add deadline +' : this.state.currentTask.deadline}</Text>
                </TouchableOpacity>
                <DateTimePicker
                    isVisible={this.state.isDatePickerVisible}
                    onConfirm={this._handleDatePicked}
                    onCancel={this._toggleDatePicker}
                    is24Hour={true}
                    timeZoneOffsetInMinutes={0}
                    mode={"datetime"}
                />
                <TouchableOpacity style={styles.createTaskButton} onPress={this._postTask}>
                    <Text style={styles.createTaskButtonText}>Create task âœ“</Text>
                </TouchableOpacity>
            </View>
        );
    }
}
