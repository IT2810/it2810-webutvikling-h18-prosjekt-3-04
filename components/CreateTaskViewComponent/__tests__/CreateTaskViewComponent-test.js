import React from 'react';
import CreateTaskView from '../CreateTaskView';
import renderer from 'react-test-renderer';
import {RequestPermission} from '../../../util/Permissions';
import {AddTodo} from "../../../util/AsyncStorage";

// Global variables
let component;
let getParam = jest.fn();
let navigate = jest.fn();
let navigation = {
    navigate: navigate,
    getParam: getParam,
};

// Mocks
jest.mock('../../../util/Permissions.js', () => ({
    RequestPermission: jest.fn(),
}));

jest.mock('expo', () => ({
    ImageManipulator: { // ImageManipulator should return a base64-string when finished
        manipulate: () => ({ base64: 'base64string'}),
    },
    ImagePicker: {
        launchCameraAsync: () => ({ cancelled: false }),
        launchImageLibraryAsync: () => ({ cancelled: false }),
    }
}));

jest.mock('../../../util/AsyncStorage.js', () => ({
    AddTodo: jest.fn(),
}));

// Generate fresh instance of module for each test
beforeEach(() => {
    const testCreateTaskViewComponent = <CreateTaskView navigation={navigation}/>;
    component = renderer.create(testCreateTaskViewComponent).getInstance();
});

test('renders correctly for text type', () => {
    getParam = jest.fn(x => 'text');
    navigation = {
      getParam: getParam,
    };

    const testCreateTaskViewComponent = <CreateTaskView navigation={navigation}/>;
    const tree = renderer.create(testCreateTaskViewComponent).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly for image type', () => {
    getParam = jest.fn(x => 'image');
    navigation = {
        getParam: getParam,
    };

    const testCreateTaskViewComponent = <CreateTaskView navigation={navigation}/>;
    const tree = renderer.create(testCreateTaskViewComponent).toJSON();
    expect(tree).toMatchSnapshot();
});

test('updates image correctly', async () => {
    let result = {
        cancelled: true,
    };

    // Data should not update if user does not choose image
    await component._updateImage(result);
    expect(component.state.currentTask.data).toBeNull();

    result = {
        cancelled: false,
    };

    // Data should update with base64-string if an image is chosen.
    await component._updateImage(result);
    expect(component.state.currentTask.data).toBe('data:image/jpeg;base64,base64string');
});

test('choosing image from camera runs correctly', async () => {

    // If permissions are granted, data state should update with base64-string
    RequestPermission.mockImplementation(() => true );
    await component._pickCameraImage();
    expect(component.state.currentTask.data).toBe('data:image/jpeg;base64,base64string');

    component.setState({ currentTask: {
        type: 'image',
        data: null
    }});

    // If permissions are not granted, state should not update and data should be null
    RequestPermission.mockImplementation(() => false );
    await component._pickCameraImage();
    expect(component.state.currentTask.data).toBeNull();

});

test('choosing image from library runs correctly', async () => {

    // If permissions are granted, data state should update with base64-string
    RequestPermission.mockImplementation(() => true );
    await component._pickImage();
    expect(component.state.currentTask.data).toBe('data:image/jpeg;base64,base64string');

    component.setState({ currentTask: {
            type: 'image',
            data: null
        }});

    // If permissions are not granted, state should not update and data should be null
    RequestPermission.mockImplementation(() => false );
    await component._pickImage();
    expect(component.state.currentTask.data).toBeNull();
});

test('writing text updates data state', () => {
    component._updateText('This is text');
    expect(component.state.currentTask.data).toBe('This is text');
});

test('writing steps text updates data state', () => {
    component._updateStepsText('This is step text');
    expect(component.state.currentTask.data).toBe('0/'+'This is step text');
});

test('choosing date updates deadline state', () => {
    component._handleDatePicked('Tue Oct 16 2018 19:24:08 GMT+0200 (CEST)');
    expect(component.state.currentTask.deadline).toBe('Tue Oct 16 2018 at 19:24');
});

test('toggle DateTimePicker should update visible state', () => {
    component._toggleDatePicker();
    expect(component.state.isDatePickerVisible).toBeTruthy();
    component._toggleDatePicker();
    expect(component.state.isDatePickerVisible).toBeFalsy();
});

test('toggle modal should update visible state', () => {
    component._toggleModal();
    expect(component.state.isModalVisible).toBeTruthy();
    component._toggleModal();
    expect(component.state.isModalVisible).toBeFalsy();
});

test('posting empty task should prevent storage update', async() => {
    global.alert = jest.fn().mockImplementation(() => {
        console.log('Alert called')
    });

    // Data is null when postTask is called here
    await component._postTask();
    expect(navigate).not.toBeCalled();
    expect(component.state.currentTask.data).toBeNull();
});

test('posting task should update storage and navigate to home', async() => {
    let navigation2 = {
        navigate: jest.fn(),
        getParam: jest.fn()
    };
    const testCreateTaskViewComponent = <CreateTaskView navigation={navigation2}/>;
    component = renderer.create(testCreateTaskViewComponent).getInstance();

    // Same as last text, but we update data state with text before calling postTask.
    // Navigate should be called after updating AsyncStorage
    component._updateText('This is text');
    await component._postTask();
    expect(component.state.currentTask.data).not.toBeNull();
    expect(navigation2.navigate).toBeCalled();
    expect(AddTodo).toBeCalled();
});
