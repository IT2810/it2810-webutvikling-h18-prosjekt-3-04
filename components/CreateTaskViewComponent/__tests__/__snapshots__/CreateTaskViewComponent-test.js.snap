// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for image type 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "marginTop": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#e8e8e8",
          "borderColor": "#d6d6d6",
          "borderRadius": 10,
          "borderWidth": 2,
          "display": "flex",
          "justifyContent": "center",
          "width": "90%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#d6d6d6",
            "fontSize": 24,
          }
        }
      >
        +
      </Text>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0)",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
          "width": "90%",
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "marginLeft": 15,
                "marginRight": 15,
              },
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              pointerEvents="box-only"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "backgroundColor": "#ff0042",
                    "borderRadius": 10,
                    "margin": 5,
                    "width": "100%",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Choose from library!
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "marginLeft": 15,
                "marginRight": 15,
              },
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              pointerEvents="box-only"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#9E9E9E",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 19,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "padding": 12,
                  },
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "backgroundColor": "#ff0042",
                    "borderRadius": 10,
                    "margin": 5,
                    "width": "100%",
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Take picture with camera!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#a3052e",
        "borderColor": "#a3052e",
        "borderRadius": 20,
        "borderWidth": 1,
        "elevation": 1,
        "marginRight": "5%",
        "opacity": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
          "paddingBottom": 5,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 5,
        }
      }
    >
      Add deadline +
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff0042",
        "borderColor": "#ff0042",
        "borderRadius": 20,
        "borderWidth": 1,
        "bottom": 50,
        "elevation": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
        "width": 150,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Create task ✓
    </Text>
  </View>
</View>
`;

exports[`renders correctly for text type 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    blurOnSubmit={true}
    containerViewStyle={
      Object {
        "marginLeft": -10,
        "marginRight": null,
        "width": "105%",
      }
    }
    enablesReturnKeyAutomatically={true}
    multiline={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Add some descriptive and motivating text to your task! 💪🏼"
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#f3f3f3",
        "borderColor": "#f3f3f3",
        "borderRadius": 15,
        "borderWidth": 1,
        "fontSize": 17,
        "fontWeight": "bold",
        "height": 200,
        "marginBottom": 20,
        "marginTop": 20,
        "padding": 10,
        "textAlignVertical": "top",
        "width": "90%",
      }
    }
    underlineColorAndroid="rgba(0,0,0,0)"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#a3052e",
        "borderColor": "#a3052e",
        "borderRadius": 20,
        "borderWidth": 1,
        "elevation": 1,
        "marginRight": "5%",
        "opacity": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
          "paddingBottom": 5,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 5,
        }
      }
    >
      Add deadline +
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff0042",
        "borderColor": "#ff0042",
        "borderRadius": 20,
        "borderWidth": 1,
        "bottom": 50,
        "elevation": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
        "width": 150,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Create task ✓
    </Text>
  </View>
</View>
`;
